src/main/java/org/openhds/service/impl/ProjectCodeService.java

```diff
@@ -21,11 +21,12 @@ public ProjectCodeService(ProjectCodeRepository projectCodeRepository) {
     }
 
     @Override
-    protected ProjectCode makeUnknownEntity() {
+    public ProjectCode makePlaceHolder(String id, String name) {
         ProjectCode projectCode = new ProjectCode();
-        projectCode.setCodeName("unknown");
```

Comment:
I think it should be `projectCode.setCodeName(id);`

so that the name can be unique.